macros/src/lib.rs

```diff
@@ -287,3 +288,19 @@ fn is_result_unit(output: &ReturnType) -> bool {
         },
     }
 }
+
+#[proc_macro_derive(HotPatchMigrate)]
```

Comment:
TODO: documentation

Reply:
Also added `#![warn(missing_docs)]` and simple top-level and `#[hot]` docs. Let me know if that's not needed.